### DISCUSSION TITLE
Windows Terminal Preview v1.20.10822.0

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#16935] [Announcements]
Windows Terminal Preview 1.20 needed some servicing too, so I figured... why not?

## Bug Fixes and Changes

* We've rolled back to an older compiler to get Terminal working on Intel Core 2 Duo machines again (!) (#16907)
* `ReadConsoleOutputCharacterW` no longer fails on lines containing narrow surrogate pairs (#16898)
* Auto-detected URLs and search results are no longer pushed around by wide characters (oops) (#16787) (thanks @Comzyh and @lhecker!)
* We will now display control characters literally if they were produced with the Windows Console APIs (#16825) (thanks @j4james!)
* So many of you have screenshots that contain our "set terminal as default?" banner, which we take to mean that you never noticed it. We got rid of it. Balance is restored. (#16873)
* Entering the tab switcher via the command palette no longer selects the wrong tab (#16858) (thanks @AlejandroBlanco2001!)
* All localizations are now checked into the project, so you can build any historic version with all of its original languages (#16835)

---

_Binary files inside the unpackaged distribution archive bear the version number `1.20.240322002-preview`._

### Asset Hashes

- Microsoft.WindowsTerminalPreview_1.20.10822.0_8wekyb3d8bbwe.msixbundle
   - SHA256 `2DE1BF8E66469C3C2F26F6398ED56EDBA33E6635AB227C9EF26D0ED8E39A90AA`
- Microsoft.WindowsTerminalPreview_1.20.10822.0_arm64.zip
   - SHA256 `7FAF725A1ADDB05B2B23692EF9970A2336589E6B68B6789B6E77A53A1688BA9F`
- Microsoft.WindowsTerminalPreview_1.20.10822.0_x64.zip
   - SHA256 `0CC49F67C51228F9677EC96D01F1E8C39720E0EA3A686E06FC8592A3F927C46D`
- Microsoft.WindowsTerminalPreview_1.20.10822.0_x86.zip
   - SHA256 `60EE40125419F0813B3C055555B388E3C9CE953EDF800B43EC85F331D2FA91B2`

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.20.10822.0'>Windows Terminal Preview v1.20.10822.0</a>.</em>

**Comment by CTimmerman** [comment]
Does this PowerShell support ANSI colors again? If so, how to enable them? They're currently broken in my Windows 11 Windows PowerShell Windows Terminal
Version: 1.19.10573.0:

```ps1
PS C:\Users\C> $PSStyle.OutputRendering = 'ANSI'
The property 'OutputRendering' cannot be found on this object. Verify that the property exists and can be set.
At line:1 char:1
+ $PSStyle.OutputRendering = 'ANSI'
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound
```

**Reply by zadjii-msft** [reply]
That doesn't look like a Terminal problem to me. That looks like a PowerShell issue? 
https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.psstyle.outputrendering?view=powershellsdk-7.4.0

Are you using PowerShell Core (7.0+)?

**Reply by zadjii-msft** [reply]
> Windows PowerShell

Oh yea, that'll probably explain it. "Windows PowerShell" is PowerShell 5, which looks like it didn't yet support `PSStyle.OutputRendering`

**Comment by CTimmerman** [comment]
The naming is extremely confusing. Thanks for looking into it.

I found the solution is using "\x1b[" instead of "\x9b", but then it stops working in Git Bash in VS Code while it starts working outside VS Code. See https://gist.github.com/CTimmerman/ddb19887b7f5e36eea122deb5eaf7431

**Reply by DHowett** [reply]
I think that C1 control sequences are not as widely supported as C0 ones. As well: if the terminal hosting Git Bash and VSCode's terminal do not support _C0 controls_... the answer may be different:

* Git Bash
   * If you're using mintty, this should work.
   * If you're using the Windows Console, see below.
* VSCode
   * VT support on Windows all-up requires console applications to _opt in_ by enabling the `ENABLE_VIRTUAL_TERMINAL_PROCESSING` "console mode". Terminal enables it by default as a convenience, but that does cause applications to rely on it as a shorthand. You should be able to trigger it handily from Python.

**Comment by Vamerlen** [comment]
thanks microsoft
😎😎😍
